src/order/ideal.lean

```diff
@@ -96,6 +106,13 @@ lemma proper_of_not_mem {I : ideal P} {p : P} (nmem : p ∉ I) : proper I :=
   exact nmem (set.mem_univ p),
 end⟩
 
+/-- A maximal ideal if it is maximal in the collection of proper ideals.
+  Note that we cannot use the `is_coatom` class because `P` might not have a `top` element.
+-/
+@[mk_iff] structure is_maximal (I : ideal P) : Prop :=
```

Comment:
Would it be better or worse for this to `extend` proper (versus having `proper` as a field)? Again, I'm asking out of curiosity, not requesting a code change.

Reply:
That's a good question. I believe that if we ever envision using `is_maximal` to get an instance of `proper`, then we'll have to make `is_maximal` extend `proper` due to the considerations at the ["forgetful inheritance" note](https://leanprover-community.github.io/mathlib_docs/notes.html#forgetful%20inheritance). Though I'm not 100% sure of this since I can't say I understand all the subtleties of type class design... maybe it doesn't matter for `Prop`s?

Reply:
Reading the note, I think it is better to extend is_proper. I added this to the PR.